.vsts-dotnet.yml

```diff
@@ -146,6 +146,28 @@ extends:
           inputs:
             dropName: $(VisualStudio.DropName)
 
+        # ensures we don't build and push experimental versions to official feeds as release versions
+        - task: PowerShell@2
+          displayName: 'Remove DotNetFinalVersionKind from Versions.props on experimental branch'
+          condition: startsWith(variables['Build.SourceBranch'], 'refs/heads/exp/')
+          inputs:
+            targetType: 'inline'
+            script: |
+              $versionsPath = "eng/Versions.props"
+              [xml]$xml = Get-Content $versionsPath
+
+              # Find and remove the DotNetFinalVersionKind node no matter where it is
+              $nodes = $xml.SelectNodes("//DotNetFinalVersionKind")
+              if ($nodes -and $nodes.Count -gt 0) {
+                foreach ($node in $nodes) {
+                  $node.ParentNode.RemoveChild($node) | Out-Null
+                }
+                $xml.Save($versionsPath)
```

Comment:
I don't love editing files on disk for this. Would it be possible to instead pass

`/p:DotNetFinalVersionKind=prerelease /p:PreReleaseVersionLabel=experimental` to the build command in this case?